### PR TITLE
OWLS-87963 - Update "Boot Identity Not Valid" FAQ section to include SSL and DNS related errors

### PR DESCRIPTION
**Pull request by ankedia**
Pull request for update to `Boot Identity Not Valid` FAQ section to include other possible causes such as communication failure between admin and managed servers due to SSL or DNS-related issues.

### CODE REVIEW

File: docs-source/content/faq/boot-identity-not-valid.md [hunk 1]
```diff
@@ -17,3 +17,13 @@ the domain directory, which occurs during the "introspection" phase, and when th
 Check that the user name and password credentials stored in the Kubernetes Secret referenced by `weblogicCredentialsSecret` contain the expected values for an account with administrative privilege for the WebLogic domain.
 Then [stop all WebLogic Server instances](https://oracle.github.io/weblogic-kubernetes-operator/userguide/managing-domains/domain-lifecycle/startup/#starting-and-stopping-servers)
 in the domain before restarting so that the operator will repeat its introspection and generate the corrected `boot.properties` files.
+
+If the "Boot identity not valid" error is still not resolved, Managed Server may be failing to communicate with the Administration Server for other reasons. Check carefully for any errors or exceptions logged before the "Boot identity not valid" error in the Managed Server log file and look for indications of communication failure reasons.
```
Comment (tbarnes-us) [inline docs-source/content/faq/boot-identity-not-valid.md:21]: Line 12, 'it means' --> 'it usually means'

Comment (tbarnes-us) [reply docs-source/content/faq/boot-identity-not-valid.md:21]: Rewrite 

`If the "Boot identity not valid" error is still not resolved, Managed Server may be failing to communicate with the Administration Server for other reasons. Check carefully for any errors or exceptions logged before the "Boot identity not valid" error in the Managed Server log file and look for indications of communication failure reasons.` 

as

`If the "Boot identity not valid" error is still not resolved, and the error is logged on a WebLogic Managed Server, then the server may be failing to contact with the Administration Server. Check carefully for any errors or exceptions logged before the "Boot identity not valid" error in the Managed Server log file and look for indications of communication failure.  For example:`

File: docs-source/content/faq/boot-identity-not-valid.md [hunk 2]
```diff
@@ -17,3 +17,13 @@ the domain directory, which occurs during the "introspection" phase, and when th
...
+
+If you enable SSL for the Administration Server and use the demo identity certificates, Managed Server may fail to establish an SSL connection with the Administration Server due to a hostname verification exception such as the "SSLKeyException: Hostname verification failed" error. For non-production environments, you can turn off the hostname verification by setting the "-Dweblogic.security.SSL.ignoreHostnameVerification=true" property in the Java options for starting the WebLogic Server.
```
Comment (tbarnes-us) [inline docs-source/content/faq/boot-identity-not-valid.md:23]: Put the above sentence, and the following sentence into a bulleted list that corresponds with the 'For example:' above.

Comment (tbarnes-us) [reply docs-source/content/faq/boot-identity-not-valid.md:23]: `If you enable SSL for the Administration Server and use the demo identity certificates, Managed Server may fail to establish an SSL connection with the Administration Server due to a hostname verification exception such as the "SSLKeyException: Hostname verification failed" error. For non-production environments, you can turn off the hostname verification by setting the "-Dweblogic.security.SSL.ignoreHostnameVerification=true" property in the Java options for starting the WebLogic Server.` 

-->

`If you have enabled SSL for the Administration Server default channel or have configured an admin port on the Administration Server, and the SSL port is using the default demo identity certificates, then a Managed Server may fail to establish an SSL connection to the Administration Server due to a hostname verification exception (such as a "SSLKeyException: Hostname verification failed" error). For non-production environments, you can turn off the hostname verification by setting the "-Dweblogic.security.SSL.ignoreHostnameVerification=true" property in the Java options for starting the WebLogic Server.`

File: docs-source/content/faq/boot-identity-not-valid.md [hunk 3]
```diff
@@ -17,3 +17,13 @@ the domain directory, which occurs during the "introspection" phase, and when th
...
+
+**NOTE**: Turning off hostname verification leaves WebLogic Server vulnerable to man-in-the-middle attacks. Oracle recommends leaving hostname verification on in production environments.
+
+For other SSL-related errors, make sure the Keystores and passwords are specified correctly in the Java options for starting the WebLogic Server. See [Configuring SSL](https://docs.oracle.com/en/middleware/fusion-middleware/weblogic-server/12.2.1.4/secmg/ssl.html#GUID-5274E688-51EC-4A63-A35E-FC718B35C897) for details on configuring SSL in the Oracle WebLogic Server environment.
+
+If the DNS hostname of the Administration Server can't be resolved during Managed Server startup with errors such as "java.net.UnknownHostException: domain1-admin-server", check the DNS server logs for any errors and take corrective actions to resolve those errors.
```
Comment (tbarnes-us) [inline docs-source/content/faq/boot-identity-not-valid.md:29]: , check --> , then check

File: docs-source/content/faq/boot-identity-not-valid.md [hunk 4]
```diff
@@ -10,20 +10,18 @@ description: "One or more WebLogic Server instances in my domain will not start
...
+If the "Boot identity not valid" error is still not resolved, and the error is logged on a WebLogic Managed Server, then the server may be failing to contact the Administration Server. Check carefully for any errors or exceptions logged before the "Boot identity not valid" error in the Managed Server log file and look for indications of communication failure. For example:
+- If you have enabled SSL for the Administration Server default channel or have configured an admin port on the Administration Server, and the SSL port is using the demo identity certificates, then a Managed Server may fail to establish an SSL connection to the Administration Server due to a hostname verification exception (such as the "SSLKeyException: Hostname verification failed"). For non-production environments, you can turn off the hostname verification by setting the "-Dweblogic.security.SSL.ignoreHostnameVerification=true" property in the Java options for starting the WebLogic Server.
```
Comment (rosemarymarano) [inline docs-source/content/faq/boot-identity-not-valid.md:21]: admin port -> administration port
"-Dweblogic.security.SSL.ignoreHostnameVerification=true" property -> `-Dweblogic.security.SSL.ignoreHostnameVerification=true` property (code font)

File: docs-source/content/faq/boot-identity-not-valid.md [hunk 5]
```diff
@@ -10,20 +10,18 @@ description: "One or more WebLogic Server instances in my domain will not start
...
 
-When you see these kinds of errors, it means that the user name and password provided in the `weblogicCredentialsSecret` are incorrect. Prior to operator version 2.5.0, this error could
-have also indicated that the WebLogic domain directory's security configuration files have changed in an incompatible way between when the operator scanned
+When you see these kinds of errors, it usually means that the user name and password provided in the `weblogicCredentialsSecret` are incorrect. Prior to operator version 2.5.0, this error could have also indicated that the WebLogic domain directory's security configuration files have changed in an incompatible way between when the operator scanned
 the domain directory, which occurs during the "introspection" phase, and when the server instance attempted to start. There is now a separate validation for that condition described in the [Domain secret mismatch](../domain-secret-mismatch/) FAQ entry.
 
 Check that the user name and password credentials stored in the Kubernetes Secret referenced by `weblogicCredentialsSecret` contain the expected values for an account with administrative privilege for the WebLogic domain.
```
Comment (rosemarymarano) [inline docs-source/content/faq/boot-identity-not-valid.md:16]: Kubernetes Secret referenced by `weblogicCredentialsSecret` contain -> Kubernetes Secret, referenced by `weblogicCredentialsSecret`, contain (needs commas for readability)

File: docs-source/content/faq/boot-identity-not-valid.md [hunk 6]
```diff
@@ -10,20 +10,18 @@ description: "One or more WebLogic Server instances in my domain will not start
...
+  **NOTE**: Turning off hostname verification leaves WebLogic Server vulnerable to man-in-the-middle attacks. Oracle recommends leaving hostname verification on in production environments.
 
-**NOTE**: Turning off hostname verification leaves WebLogic Server vulnerable to man-in-the-middle attacks. Oracle recommends leaving hostname verification on in production environments.
+- For other SSL-related errors, make sure the Keystores and passwords are specified correctly in the Java options for starting the WebLogic Server. See [Configuring SSL](https://docs.oracle.com/en/middleware/fusion-middleware/weblogic-server/12.2.1.4/secmg/ssl.html#GUID-5274E688-51EC-4A63-A35E-FC718B35C897) for details on configuring SSL in the Oracle WebLogic Server environment.
```
Comment (rosemarymarano) [inline docs-source/content/faq/boot-identity-not-valid.md:25]: Keystores  -> keystores

**Review by rosemarymarano — COMMENTED** [review]
few minor edits

**Review by rosemarymarano — APPROVED** [review]

**Comment by tbarnes-us** [comment]
LGTM now, but am starting a slack thread about a potential additional bullet.

**Review by tbarnes-us — APPROVED** [review]

**Pull request merged**